src/service/viewer-impl.js

```diff
@@ -983,7 +997,12 @@ export class Viewer {
       const queue = this.messageQueue_.slice(0);
       this.messageQueue_ = [];
       queue.forEach(message => {
-        this.messageDeliverer_(message.eventType, message.data, false);
+        const responsePromise = this.messageDeliverer_(
+            message.eventType, message.data, message.awaitResponse);
+
+        if (message.awaitResponse) {
+          responsePromise.then(response => message.responseResolver(response));
```

Comment:
You can optimize this by resolving with the promise: `message.responseResolver(responsePromise)`